src/test/java/com/jadajabajava/providers/HhVacancyProviderTest.java

```diff
@@ -0,0 +1,163 @@
+package com.jadajabajava.providers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jadajabajava.deserializers.HhVacancyDeserializer;
+import com.jadajabajava.entities.Vacancy;
+import com.jadajabajava.exceptions.VacancyProviderException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.context.MessageSource;
+import org.springframework.http.HttpStatus;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class HhVacancyProviderTest {
+
+    @Mock
+    private HttpClient httpClient;
+
+    @Mock
+    private HttpResponse<String> httpResponseList;
+
+    @Mock
+    private HttpResponse<String> httpResponseVacancy;
+
+    @Mock
+    private ObjectMapper mapper;
+
+    @Mock
+    private MessageSource messageSource;
+
+    @Mock
+    private HhVacancyDeserializer vacancyDeserializer;
+
+    @InjectMocks
+    private HhVacancyProvider vacancyProvider;
+
+    @Captor
+    private ArgumentCaptor<HttpRequest> requestCaptor;
+
+    private static final String VACANCIES_LIST_VALID_JSON_FILE_PATH = "json/vacancies_list_valid.json";
+    private static final String VACANCIES_LIST_INVALID_JSON_FILE_PATH = "json/vacancies_list_invalid.json";
+    private static final String VACANCIES_LIST_EMPTY_JSON_FILE_PATH = "json/vacancies_list_empty.json";
+    private static final String VACANCY_VALID_JSON_FILE_PATH = "json/vacancy_valid.json";
+
+    private final ClassLoader classLoader = this.getClass().getClassLoader();
+
+    // handle invalid response
+    // handle valid response
+
+    @Test
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/jada-jaba-java/job-seeker-service/pullRequest?prid=5720132)